core/client/src/main/java/alluxio/client/AbstractOutStream.java

```diff
@@ -28,12 +28,20 @@
   // requires checking the codebase for this assumption (using int to denote an offset).
   // See ALLUXIO-1536.
   /** The number of bytes written. */
-  protected int mBytesWritten = 0;
+  protected long mBytesWritten = 0;
 
   /**
    * @return the number of bytes written to this stream
+   * @deprecated use {@link #getTotalBytesWritten()} to avoid integer overflow
    */
+  @Deprecated
   public int getBytesWritten() {
+    // Cast to int to preserve backwards compatibility. In 2.0 we should change this to long so that
+    // it can be correct.
+    return (int) mBytesWritten;
+  }
```

Comment:
May want to avoid overflow in general matching the behavior in classes like `DataOutputStream`. The behavior won't be the same, but I don't think anyone is relying on this returning a negative value.